src/jit/lclvars.h

```diff
@@ -0,0 +1,750 @@
+                                     // global location, etc.
+                                     // We cannot reason reliably about the value of the variable.
+    unsigned char lvDoNotEnregister : 1; // Do not enregister this variable.
+    unsigned char lvFieldAccessed : 1;   // The var is a struct local, and a field of the variable is accessed.  Affects
+                                         // struct promotion.
+
+#ifdef DEBUG
+    // These further document the reasons for setting "lvDoNotEnregister".  (Note that "lvAddrExposed" is one of the
+    // reasons;
+    // also, lvType == TYP_STRUCT prevents enregistration.  At least one of the reasons should be true.
+    unsigned char lvVMNeedsStackAddr : 1; // The VM may have access to a stack-relative address of the variable, and
+                                          // read/write its value.
+    unsigned char lvLiveInOutOfHndlr : 1; // The variable was live in or out of an exception handler, and this required
+                                          // the variable to be
+                                          // in the stack (at least at those boundaries.)
+    unsigned char lvLclFieldExpr : 1;     // The variable is not a struct, but was accessed like one (e.g., reading a
+                                          // particular byte from an int).
+    unsigned char lvLclBlockOpAddr : 1;   // The variable was written to via a block operation that took its address.
+    unsigned char lvLiveAcrossUCall : 1;  // The variable is live across an unmanaged call.
+#endif
+    unsigned char lvIsCSE : 1;       // Indicates if this LclVar is a CSE variable.
+    unsigned char lvRefAssign : 1;   // involved in pointer assignment
+    unsigned char lvHasLdAddrOp : 1; // has ldloca or ldarga opcode on this local.
+    unsigned char lvStackByref : 1;  // This is a compiler temporary of TYP_BYREF that is known to point into our local
+                                     // stack frame.
+
+    unsigned char lvHasILStoreOp : 1;         // there is at least one STLOC or STARG on this local
+    unsigned char lvHasMultipleILStoreOp : 1; // there is more than one STLOC on this local
+
+    unsigned char lvIsTemp : 1; // Short-lifetime compiler temp (if lvIsParam is false), or implicit byref parameter
+                                // (if lvIsParam is true)
+#if OPT_BOOL_OPS
+    unsigned char lvIsBoolean : 1; // set if variable is boolean
+#endif
+    unsigned char lvRngOptDone : 1; // considered for range check opt?
+    unsigned char lvLoopInc : 1;    // incremented in the loop?
+    unsigned char lvLoopAsg : 1;    // reassigned  in the loop (other than a monotonic inc/dec for the index var)?
+    unsigned char lvArrIndx : 1;    // used as an array index?
+    unsigned char lvArrIndxOff : 1; // used as an array index with an offset?
+    unsigned char lvArrIndxDom : 1; // index dominates loop exit
```

Comment:
Since you're here, it would be good to be consistent with describing this booleans. To me, when the comment has a "?" at the end it sounds as if the author wasn't sure. I would either phrase it as an assertion, like this one "index dominates loop exit" or as a complete question "Does the index dominate the loop exit".